tracing-appender/src/rolling.rs

```diff
@@ -1539,49 +1228,109 @@ mod test {
     }
 
     #[test]
-    fn size_based_rotation() {
+    fn test_max_log_size() {
+        use std::sync::{Arc, Mutex};
+        use tracing_subscriber::prelude::*;
+
+        let format = format_description::parse(
+            "[year]-[month]-[day] [hour]:[minute]:[second] [offset_hour \
+         sign:mandatory]:[offset_minute]:[offset_second]",
+        )
+            .unwrap();
+
+        let now = OffsetDateTime::parse("2020-02-01 10:01:00 +00:00:00", &format).unwrap();
         let directory = tempfile::tempdir().expect("failed to create tempdir");
+        let (state, writer) = Inner::new(
+            now,
+            Rotation::HOURLY.with_max_bytes(10),
+            directory.path(),
+            None,
+            None,
+            None,
+        )
+            .unwrap();
 
-        let mut appender = RollingFileAppender::builder()
-            .rotation(Rotation::max_bytes(8))
-            .filename_prefix("size_based_rotation")
-            .filename_suffix("log")
-            .build_with_now(&directory, Box::new(|| OffsetDateTime::UNIX_EPOCH))
-            .expect("initializing rolling file appender failed");
+        let clock = Arc::new(Mutex::new(now));
+        let now = {
+            let clock = clock.clone();
+            Box::new(move || *clock.lock().unwrap())
+        };
+        let appender = RollingFileAppender { state, writer, now };
+        let default = tracing_subscriber::fmt()
```

Comment:
naming: I feel it’s more readable named like this (`default` could be many things)
```suggestion
        let subscriber = tracing_subscriber::fmt()
```